### DISCUSSION TITLE
Using TWMs for material decay?

### DISCUSSION THREAD

**Discussion by wizardofgcc** [jeremyshannon/Exile#108] [Ideas]
So I was looking at the dependencies of naturalslopeslib, and found that it can use [twmlib](https://content.minetest.net/packages/karamel/twmlib/). This library allows registering a more limited abm-like process, which is likely to have a better perofrmance (though I have not fully verified this yet). This mod could be used to implement material decay, e. g. rammed earth eroding or logs rotting, with a better potential performance than ABMs. Maybe also useful for things like soaking of soil? That seems to be a performance drain, currently.

**Comment by jeremyshannon** [comment]
Good catch! This could be worth investigating.
